pkg/plugin/block_store.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2017 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"github.com/hashicorp/go-plugin"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/heptio/ark/pkg/cloudprovider"
+	proto "github.com/heptio/ark/pkg/plugin/generated"
+)
+
+// BlockStorePlugin is an implementation of go-plugin's Plugin
+// interface with support for gRPC for the cloudprovider/BlockStore
+// interface.
+type BlockStorePlugin struct {
+	plugin.NetRPCUnsupportedPlugin
+
+	impl cloudprovider.BlockStore
+}
+
+// NewBlockStorePlugin constructs a BlockStorePlugin.
+func NewBlockStorePlugin(blockStore cloudprovider.BlockStore) *BlockStorePlugin {
+	return &BlockStorePlugin{
+		impl: blockStore,
+	}
+}
+
+// GRPCServer registers a BlockStore gRPC server.
+func (p *BlockStorePlugin) GRPCServer(s *grpc.Server) error {
+	proto.RegisterBlockStoreServer(s, &BlockStoreGRPCServer{impl: p.impl})
+	return nil
+}
+
+// GRPCClient returns a BlockStore gRPC client.
+func (p *BlockStorePlugin) GRPCClient(c *grpc.ClientConn) (interface{}, error) {
+	return &BlockStoreGRPCClient{grpcClient: proto.NewBlockStoreClient(c)}, nil
+}
+
+// BlockStoreGRPCClient implements the cloudprovider.BlockStore interface and forwards
+// calls to a gRPC client.
```

Comment:
It forwards to a gRPC server, right?

Reply:
this struct makes calls on the gRPC client, which get routed to the gRPC server (plugin). Maybe "forward" is a bad word? Could say "...uses a gRPC client to make calls"?